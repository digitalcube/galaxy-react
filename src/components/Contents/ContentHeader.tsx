import React, { FC } from 'react';

export type ContentHeaderProps = {
  href: string;
  src: string;
  width: string;
  height: string;
  alt?: string;
  title: string;
};
export const ContentHeader: FC<ContentHeaderProps> = ({
  href,
  src,
  width,
  height,
  alt,
  title,
}) => {
  return (
    <div className="shifter-guides">
      <h1 className="guide-heading-1 font-weight-bold">{title}</h1>
      {src ? (
        <div className="mt-4 container-fluid">
          <div className="row">
            <div className="col px-0">
              <a
                href={href}
                rel="noopener noreferrer"
                target="_blank"
                className="d-block"
              >
                <img
                  className="img-fluid"
                  src={src}
                  width={width}
                  height={height}
                  alt={alt}
                />
              </a>
            </div>
          </div>
        </div>
      ) : null}
    </div>
  );
};
