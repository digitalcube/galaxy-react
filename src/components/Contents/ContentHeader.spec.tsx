import { render } from '@testing-library/react';
import React from 'react';
import { Default as ContentHeader } from './ContentHeader.stories';

describe('ContentHeader', () => {
  it('to match snapshot [column %p | item %p]', () => {
    const { asFragment } = render(
      <ContentHeader {...(ContentHeader.args as any)} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
