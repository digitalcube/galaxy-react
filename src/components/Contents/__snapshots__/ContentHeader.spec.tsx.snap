// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentHeader to match snapshot [column %p | item %p] 1`] = `
<DocumentFragment>
  <div
    class="shifter-guides"
  >
    <h1
      class="guide-heading-1 font-weight-bold"
    >
      Guides
    </h1>
    <div
      class="mt-4 container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="col px-0"
        >
          <a
            class="d-block"
            href="#"
            rel="noopener"
            target="_blank"
          >
            <img
              alt="Introducing PHP Error Logs"
              class="img-fluid"
              height="180"
              src="/images/guide-banner.png"
              width="1134"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
